srcpkgs/python3-py7zr/template

```diff
@@ -0,0 +1,16 @@
+# Template file for 'python3-py7zr'
+pkgname=python3-py7zr
+version=0.11.3
+revision=1
+wrksrc="py7zr-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-pip python3-pycryptodome python3-pyannotate"
```

Comment:
We shouldn't have pip here, can you show the error you got without it?

Reply:
```
=> python3-py7zr-0.11.3_1: running do_build ...
/usr/lib/python3.9/site-packages/setuptools/dist.py:634: UserWarning: Usage of dash-separated 'max-line-length' will not be supported in future versions. Please use the underscore name 'max_line_length' instead
  warnings.warn(
WARNING: The wheel package is not available.
/usr/bin/python3: No module named pip
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/setuptools/installer.py", line 75, in fetch_build_egg
    subprocess.check_call(cmd)
  File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmphi1lzhxq', '--quiet', 'pep517']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/builddir/py7zr-0.11.3/setup.py", line 5, in <module>
    setup(use_scm_version={"local_scheme": "no-local-version"})
  File "/usr/lib/python3.9/site-packages/setuptools/__init__.py", line 152, in setup
    _install_setup_requires(attrs)
  File "/usr/lib/python3.9/site-packages/setuptools/__init__.py", line 147, in _install_setup_requires
    dist.fetch_build_eggs(dist.setup_requires)
  File "/usr/lib/python3.9/site-packages/setuptools/dist.py", line 713, in fetch_build_eggs
    resolved_dists = pkg_resources.working_set.resolve(
  File "/usr/lib/python3.9/site-packages/pkg_resources/__init__.py", line 766, in resolve
    dist = best[req.key] = env.best_match(
  File "/usr/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1051, in best_match
    return self.obtain(req, installer)
  File "/usr/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1063, in obtain
    return installer(requirement)
  File "/usr/lib/python3.9/site-packages/setuptools/dist.py", line 772, in fetch_build_egg
    return fetch_build_egg(self, req)
  File "/usr/lib/python3.9/site-packages/setuptools/installer.py", line 77, in fetch_build_egg
    raise DistutilsError(str(e)) from e
distutils.errors.DistutilsError: Command '['/usr/bin/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmphi1lzhxq', '--quiet', 'pep517']' returned non-zero exit status 1.
=> ERROR: python3-py7zr-0.11.3_1: do_build: 'python3 setup.py build ${make_build_args}' exited with 1
=> ERROR:   in do_build() at common/build-style/python3-module.sh:22
```

Reply:
My guess is that it can't find `wheel` so it fallbacks to use `pip` to install `wheel`.
You can try replacing `python3-setuptools` with `python3-wheel`, maybe it will work

Reply:
i tried by doesn't work

Reply:
```
install_requires =
      texttable
      pycryptodome
      importlib_metadata;python_version<"3.8"
      pyzstd>=0.14.4,<0.15.0
      ppmd-cffi>=0.3.1,<0.4.0
      bcj-cffi>=0.5.0,<0.6.0
      multivolumefile>=0.2.0,<0.3.0
setup_requires =
      setuptools-scm>=3.5.0
      setuptools>=42.0
```

Try adding `python3-setuptools_scm` (and remove `python3-pip`)

Reply:
stil same

Reply:
You can't replace `python3-setuptools` with `python3-wheel`. You need `python3-setuptools_scm` and `python3-wheel` together. Drop `python3-pip` and you can also drop `python3-setuptools` because `setuptools_scm` will pull it in. Be on the lookout for errors related to `toml`, because some packages that use `setuptools_scm` also require `python3-toml`. (Don't add that unless it's really necessary.)

Reply:
tried `hostmakedepends="python3-setuptools_scm python3-wheel python3-pycryptodome python3-pyannotate"` but still get the pip error..

Reply:
https://github.com/miurahr/py7zr/blob/2b082f41bd83022a89cd35712840170f137dff0a/pyproject.toml#L2 suggests that you need to add the `python3-toml` dependency I mentioned to `hostmakedepends`.